ariadne/wsgi_middleware.py

```diff
@@ -0,0 +1,160 @@
+        data = self.get_request_data(environ)
+        result = self.execute_query(environ, data)
+        return self.return_response_from_result(start_response, result)
+
+    def get_request_data(self, environ: dict) -> dict:
+        if environ["CONTENT_TYPE"] != CONTENT_TYPE_JSON:
+            raise HttpBadRequestError(
+                "Posted content must be of type {}".format(CONTENT_TYPE_JSON)
+            )
+
+        request_content_length = self.get_request_content_length(environ)
+        request_body = environ["wsgi.input"].read(request_content_length)
+        if not request_body:
+            raise HttpBadRequestError("request body cannot be empty")
+
+        data = self.parse_request_body(request_body)
+        if not isinstance(data, dict):
+            raise HttpBadRequestError("valid request body should be a JSON object")
+
+        return data
+
+    def parse_request_body(self, request_body: bytes) -> Any:
+        try:
+            return json.loads(request_body)
+        except (TypeError, ValueError):
+            raise HttpBadRequestError("request body is not a valid JSON")
+
+    def get_request_content_length(self, environ: dict) -> int:
+        try:
+            return int(environ.get("CONTENT_LENGTH", 0))
+        except (TypeError, ValueError):
+            raise HttpBadRequestError("content length header is missing or incorrect")
+
+    def execute_query(self, environ: dict, data: dict) -> ExecutionResult:
+        return graphql(
+            self.schema,
+            data.get("query"),
+            root=self.get_query_root(environ, data),
+            context=self.get_query_context(environ, data),
+            variables=data.get("variables"),
+            operation_name=data.get("operationName"),
+        )
+
+    def get_query_root(
+        self, environ: dict, request_data: dict  # pylint: disable=unused-argument
+    ) -> Any:
+        """Override this method in inheriting class to create query root."""
+        return None
+
+    def get_query_context(
+        self, environ: dict, request_data: dict  # pylint: disable=unused-argument
+    ) -> Any:
+        """Override this method in inheriting class to create query context."""
+        return {"environ": environ}
+
+    def return_response_from_result(
+        self, start_response: Callable, result: ExecutionResult
+    ) -> List[bytes]:
+        status = HTTP_STATUS_200_OK
+        response = {}
+        if result.errors:
+            response["errors"] = [format_error(e) for e in result.errors]
+        if result.invalid:
+            status = HTTP_STATUS_400_BAD_REQUEST
+        else:
+            response["data"] = result.data
+
+        start_response(status, [("Content-Type", CONTENT_TYPE_JSON)])
+        return [json.dumps(response).encode("utf-8")]
+
+    @classmethod
+    def make_simple_server(
+        cls,
+        type_defs: Union[str, List[str]],
+        resolvers: Union[dict, List[dict]],
+        port: int = 8888,
```

Comment:
Why not allowing to pass the host? 
Also, maybe we could add a little docstring with an example usage? e.g.:

```python
>>> type_defs = """
...    type Query {
...        test: String
...    }
... """
>>> host = "0.0.0.0"
>>> port = 8888
>>>
>>> playground_httpd = GraphQLMiddleware.make_simple_server(
...     type_defs=type_defs, resolvers={}, host=host, port=port)
...
>>> print(f'Now serving on http://{host}:{port}')
>>> playground_httpd.serve_forever()
>>> 
```

I know it's kinda temporary and hacky PR! Awesome work anyway :+1:

Reply:
For a development server, I can't see a real difference unlike a WSGI production server.

Apart that I think it is odd that we are making the development server to listen on the local network interface (`0.0.0.0`) instead of the `loopback` interface. As this would expose the development application to the rest of the local network (unless the firewall is blocking it of course), I feel like it would be an unwanted behavior.

Maybe we should default it `127.0.0.1` and allow the user to set the listening host to `0.0.0.0` if the user explicitly wants to?

Or am I missing a reason that the server is being defaulted (or hardcoded) to `0.0.0.0`?

Reply:
Letting user specify the host, even if its `0.0.0.0` vs default `127.0.0.1` is good enough reason to make host configurable 👍

I'll change that, thanks!